SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -1199,10 +1199,23 @@ void USpatialNetDriver::OnOwnerUpdated(AActor* Actor, AActor* OldOwner)
 
 void USpatialNetDriver::ProcessOwnershipChanges()
 {
+	const bool bShouldWriteLoadBalancingData = GetDefault<USpatialGDKSettings>()->bEnableStrategyLoadBalancingComponents;
+	TOptional<const SpatialGDK::LoadBalancingWriter> LoadBalancingWriter;
+
+	if (bShouldWriteLoadBalancingData)
+	{
+		LoadBalancingWriter.Emplace(this);
+	}
+
 	for (Worker_EntityId EntityId : OwnershipChangedEntities)
 	{
 		if (USpatialActorChannel* Channel = GetActorChannelByEntityId(EntityId))
 		{
+			if (bShouldWriteLoadBalancingData)
```

Comment:
I'd check the optional is valid here, or, check the bool but assert if the optional is invalid?

Reply:
Do you think there's benefit in this? Both the flag and the optional are set and created in the same function, I don't see how they can become out of sync.